.github/actions/setup-etcd/action.yml

```diff
@@ -0,0 +1,34 @@
+name: 'Setup etcd'
+description: 'Setup etcd and add it to the PATH'
+inputs:
+  etcd-version:
+    description: 'Release name from https://github.com/etcd-io/etcd/releases'
+    required: true
+    default: v3.4.13
+  install-prefix:
+    description: 'Where to extract the archive'
+    default: './etcd'
+runs:
+  using: 'composite'
+  steps:
+    - shell: bash
+      env:
+        BASE_URL: "https://github.com/etcd-io/etcd/releases/download"
+        ETCD_VER: ${{ inputs.etcd-version }}
+        INSTALL_PREFIX: ${{ inputs.install-prefix }}
+      run: >
+        set -x;
```

Comment:
fyi: you can specify bash options right in the shell keyword:

```yaml
shell: bash --noprofile --norc -eox pipefail {0}
```

Reply:
Ok, but it doesn't make things simpler, so I'd rather avoid it.